### PR TITLE
Source Edit modal rebuilding

### PR DESCRIPTION
**Pull request by rvsia**
https://projects.engineering.redhat.com/browse/TPINVTRY-703

**Description**

- Rebuilds the Edit form to modal

image

editform

TODO: 
- [ ] ~~load billing_source info (no GET endpoint)~~ (will do in a different PR, this is too big)
- [ ] ~~push billing_source info (no PATCH endpoint)~~ (will do in a different PR, this is too big)
- [x] length validators https://github.com/data-driven-forms/react-forms/issues/231
- [x] tests
- [ ] ~~jump off imported source (not editable)~~ (will do in a different PR, this is too big)
- [x] https://github.com/RedHatInsights/frontend-components/pull/274
- [x] internationalization
- [x] handle submit values

### CODE REVIEW

**Comment by rvsia** [comment]
@Hyperkid123 Please review.. and don't worry, most files are tests. :smile_cat:

File: src/api/entities.js [hunk 1]
```diff
@@ -28,71 +29,26 @@ export function doRemoveSource(sourceId) {
...
-        getSourcesApi().updateEndpoint(source.endpoint.id, endpointData).catch((error) => {
-            throw { error: { title: errorTitles.endpoint, detail: error.errors[0].detail } };
-        }),
-        getSourcesApi().updateAuthentication(source.authentication.id, formData.authentication).catch((error) => {
-            throw { error: { title: errorTitles.authentication, detail: error.errors[0].detail } };
-        })
-    ]);
-};
+    return getSourcesApi().listEndpointAuthentications(endpoint.id).then(authentications => {
+        return ({
```
Comment (Hyperkid123) [inline src/api/entities.js]: no block

File: src/components/SourceEditForm/parser/application.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    .map((key) => authentication[key].fields.filter(({ name }) => name.startsWith('billing_source')))
+    .flatMap((x) => x);
+
+export const getEnhancedBillingSourceField = (sourceType, name, authenticationsTypes) => {
+    let field = undefined;
+
+    authenticationsTypes.forEach((type) => {
+        const apps = field ? {} : get(hardcodedSchemas, [sourceType, 'authentication', type], {});
+
+        Object.keys(apps).forEach((key) => {
```
Comment (Hyperkid123) [inline src/components/SourceEditForm/parser/application.js]: use find

File: src/components/SourceEditForm/parser/application.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        title: costManagementApp.display_name,
+        name: costManagementApp.display_name,
+        fields: modifyFields(enhandcedFields, editing, setEdit)
+    });
+};
+
+export const applicationsFields = (applications, sourceType, editing, setEdit, appTypes, authentications) => {
+    const fields = [];
+
+    fields.push(costManagementFields(
```
Comment (Hyperkid123) [inline src/components/SourceEditForm/parser/application.js]: return just this

File: src/components/SourceEditForm/parser/authentication.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { hardcodedSchemas } from '@redhat-cloud-services/frontend-components-sources';
+import { EDIT_FIELD_NAME } from '../../editField/EditField';
+import { FormattedMessage } from 'react-intl';
+
+export const createAuthFieldName = (fieldName, id) => `authentications.a${id}.${fieldName.replace('authentication.', '')}`;
+
+export const getLastPartOfName = (fieldName) => fieldName.split('.').pop();
+
+export const removeRequiredValidator = (validate = []) =>
+    validate.map(validation => validation.type === validatorTypes.REQUIRED ? undefined : validation)
```
Comment (Hyperkid123) [inline src/components/SourceEditForm/parser/authentication.js]: use `filter`

File: src/components/SourceEditForm/parser/parseSourceToSchema.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { genericInfo } from './genericInfo';
+import { authenticationFields } from './authentication';
+import { endpointFields } from './endpoint';
+import { applicationsFields } from './application';
+
+export const parseSourceToSchema = (source, editing, setEdit, sourceType, appTypes) => {
+    const fields = [];
+
+    fields.push(...genericInfo(editing, setEdit, source.source.id));
```
Comment (Hyperkid123) [inline src/components/SourceEditForm/parser/parseSourceToSchema.js]: No need for pushing. Just put it into the array

**Comment by rvsia** [comment]
@Hyperkid123 updated

**Pull request merged**